lib/logstash/filters/elasticsearch.rb

```diff
@@ -103,8 +103,14 @@ class LogStash::Filters::Elasticsearch < LogStash::Filters::Base
   # Comma-delimited list of `<field>:<direction>` pairs that define the sort order
   config :sort, :validate => :string, :default => "@timestamp:desc"
 
-  # Array of fields to copy from old event (found via elasticsearch) into new event
-  config :fields, :validate => :array, :default => {}
+  # Hash of fields to copy from old event (found via elasticsearch) into new event
+  config :fields, :validate => :hash, :default => {}
```

Comment:
this represents a breaking change, the following config which worked before will not work anymore:

```
  filter {
    elasticsearch {
      fields => [["response", "code"]]
      #  ...
    }
  }
```

We can either revert this change or bump the plugin to 4.0.0, but since breaking changes are so costly when it comes to maintenance I'd suggest reverting. thoughts?

Reply:
Indeed, it is better to avoid breaking changes.
I did it to be consistent with actual plugin documentation (and the implementation code), which both uses this option as a hash (not as an array).
But ok, I would revert it to array.